ingeniamotion/pdo.py

```diff
@@ -194,17 +194,50 @@ def __init__(
                 )
             self._notify_send_process_data = notify_send_process_data
             self._notify_receive_process_data = notify_receive_process_data
+            self._notify_exceptions = notify_exceptions
 
         def run(self) -> None:
             """Start the PDO exchange"""
-            self._net.start_pdos()
+            try:
+                self._net.start_pdos()
+            except (ILStateError, ILError) as il_error:
+                self._pd_thread_stop_event.set()
+                im_exception = IMException(
+                    f"Could not start the PDOs due to the following exception: {il_error}"
+                )
+                self._notify_exceptions(im_exception)
+            iteration_duration: float = -1
             while not self._pd_thread_stop_event.is_set():
```

Comment:
If there's an exception during the initialization I think the process data loop should not start

Reply:
It will not start because the the `_pd_thread_stop_event` is set.